ooniprobe/View/TestResults/Details/TestDetailsViewController.m

```diff
@@ -49,7 +49,13 @@ - (void)advancedScreens{
                                    actionWithTitle:NSLocalizedString(@"TestResults.Details.RawData", nil)
                                    style:UIAlertActionStyleDefault
                                    handler:^(UIAlertAction * action) {
-                                       [self rawData];
+                                       if (![[[ReachabilityManager sharedManager] getStatus] isEqualToString:@"no_internet"]
+                                           && isInExplorer)
```

Comment:
I don't get why `isInExplorer` is checked for here

Reply:
Because otherwise we load the text file from disk, so there is no need to check connection

Reply:
Can you write an utility function to encapsulate the "there is no internet" logic? I was confused.

My understanding is that the logic is like: (1) if we have internet access and the data is in explorer, then return the data (?) otherwise (2) show an error. Do I understand it correctly?

(This seems also a good moment to say again that

```
if (!condition) {
  error();
  return;
}
// ...
```

is generally more readable for me and less complex to parse than structured programming.)

Reply:
There is no internet logic is `[[[ReachabilityManager sharedManager] getStatus] isEqualToString:@"no_internet"]`
if you want I can write a feature that incapsulate this code

Reply:
we can't do the code format you wrote as we need to execute different code if there is internet or not so I need an if else

Reply:
The logic is: if there is no internet and the measurement is in explorer (so it's not in local) show an error.
Else go to the rawData screen that will handle the logic (load from disk or from internet)

Reply:
> if you want I can write a feature that incapsulate this code

Yes, please!

Reply:
@nuke what I am saying is that you can write (in pseudo JavaScript):

```javascript
function () {
  if (!haveInternet() || !haveMeasurement) {
    showError();
    return;
  }
  showMeasurement();
}
```